### PR TITLE
UAR-1432 Only check Trust Ceased Date when full validation running

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-1432

### Change description

* Can't checked Ceased Date when running partial validation as Mongo data might be inconsistent (until all web validation on all screens has run)
* Simple check implemented
* Unit tests added and amended

### Work checklist

- [X] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustDetailsValidator.java [hunk 1]
```diff
@@ -23,7 +23,10 @@
 @Component
 public class TrustDetailsValidator {
 
-    public Errors validate(OverseasEntitySubmissionDto overseasEntitySubmissionDto, Errors errors, String loggingContext) {
+    public Errors validate(OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+                           Errors errors,
+                           String loggingContext,
+                           boolean checkCeasedDate) {
```
Comment (dlloyd1-ch) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustDetailsValidator.java:29]: Method is called with the boolean variable isFullValidation mapped to checkCeasedDate. Might be more readable if isFullValidation is used throughout.

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java [hunk 2]
```diff
@@ -322,7 +322,7 @@ void testEntityNameFieldValidatorGetsCalled() {
         verify(presenterDtoValidator, times(0)).validate(any(), any(), any());
         verify(entityDtoValidator, times(0)).validate(any(), any(), any());
         verify(dueDiligenceDataBlockValidator, times(1)).validatePartialDueDiligenceFields(any(), any(), any(), any());
-        verify(trustDetailsValidator, times(0)).validate(any(), any(), any());
+        verify(trustDetailsValidator, times(0)).validate(any(), any(), any(), eq(true));
```
Comment (dlloyd1-ch) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:325]: validatePartial is being called in this test so not sure why isFullValidation would be true here.

**Review by dlloyd1-ch — COMMENTED** [review]
Couple of minor comments

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java (hunk 2 above)
Comment (mwestacott) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:325]: Good spot. It shouldn't be called at all, with either a true or a false value. Will fix.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustDetailsValidator.java (hunk 1 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustDetailsValidator.java:29]: Yeah. I can do that. Will change the parameter name.

**Review by dlloyd1-ch — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**